### PR TITLE
Show count in Results, Files and Logs tab, display message if no result is found and add highlight the critical files with an icon.

### PR DESCRIPTION
**Pull request by tushar-rishav**
Changes added:
 * Display count for various tabs.
 * Sort the file hierarchy - makes navigation easier.
 * Rather than showing a blank space, add a suitable message in case of no results/logs are found.
 * Display n suitable icon along with the file name for the files that produced results.
 * Enable user to skip multiple directories while navigating. Clicking on individual level will navigate to that level.
See the
[demo](https://cloud.githubusercontent.com/assets/7397433/15913930/cf8d2342-2dfa-11e6-85e1-968f042dd69a.gif)

### CODE REVIEW

File: coalahtml/_coalahtml/app/app.js [hunk 1]
```diff
@@ -2,12 +2,35 @@
 
 angular
   .module('coalaHtmlApp', ['ngRoute', 'ngSanitize'])
-  .run(function($rootScope) {
+  .run(function($rootScope, $http) {
```
Comment (Uran198) [inline coalahtml/_coalahtml/app/app.js]: I'm not the Angular expert, but I think it may be better to inject arguments with list notation. This way you'll be able to minify your js files without problems and it'll be easier to test with dependency injections. Please correct me, if I'm missing something.

File: coalahtml/_coalahtml/app/app.js [hunk 2]
```diff
@@ -2,12 +2,35 @@
...
     $rootScope.LOG_LEVELS = ["DEBUG", "INFO", "WARNING", "ERROR"];
     $rootScope.RESULT_SEVERITY = ["INFO", "NORMAL", "MAJOR"];
     $rootScope.SEVERITY_TO_BOOTSTRAP = ["info", "warning", "danger"];
     $rootScope.LOG_LEVEL_TO_BOOTSTRAP = ["primary", "info", "warning",
                                          "danger"];
+    var getData = function (url) {
+      return $http.get(url);
+    };
```
Comment (Uran198) [inline coalahtml/_coalahtml/app/app.js]: How is this better that raw `$http.get(url)`?

Comment (tushar-rishav) [inline coalahtml/_coalahtml/app/app.js]: So this is synchronous. It returns a promise actually.
ᐧ

On Fri, Jun 3, 2016 at 2:55 PM, Attila <notifications@github.com> wrote:

> In coalahtml/_coalahtml/app/app.js
> <https://github.com/coala-analyzer/coala-html/pull/52#discussion_r65679661>
> :
>
> >      $rootScope.LOG_LEVELS = ["DEBUG", "INFO", "WARNING", "ERROR"];
> >      $rootScope.RESULT_SEVERITY = ["INFO", "NORMAL", "MAJOR"];
> >      $rootScope.SEVERITY_TO_BOOTSTRAP = ["info", "warning", "danger"];
> >      $rootScope.LOG_LEVEL_TO_BOOTSTRAP = ["primary", "info", "warning",
> >                                           "danger"];
> > +    var getData = function (url) {
> > +      return $http.get(url);
> > +    };
>
> How is this better that raw $http.get(url)?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-html/pull/52/files/b7f706d2171c77e341e20b5b680c0c87b1b38b7e#r65679661>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHDgOYIINWsziJFcvJgu0GgE60gQeRhgks5qH_L9gaJpZM4IslPq>
> .
>



-- 
Best regards,
Tushar Gautam
http://gtushar.co

Comment (tushar-rishav) [inline coalahtml/_coalahtml/app/app.js]: I needed to have those data files loaded, before I could actually use them
later. So I made it synchronous.
ᐧ

On Fri, Jun 3, 2016 at 3:06 PM, Tushar Gautam <tushar.rishav@gmail.com>
wrote:

> So this is synchronous. It returns a promise actually.
> ᐧ
>
> On Fri, Jun 3, 2016 at 2:55 PM, Attila <notifications@github.com> wrote:
>
>> In coalahtml/_coalahtml/app/app.js
>> <https://github.com/coala-analyzer/coala-html/pull/52#discussion_r65679661>
>> :
>>
>> >      $rootScope.LOG_LEVELS = ["DEBUG", "INFO", "WARNING", "ERROR"];
>> >      $rootScope.RESULT_SEVERITY = ["INFO", "NORMAL", "MAJOR"];
>> >      $rootScope.SEVERITY_TO_BOOTSTRAP = ["info", "warning", "danger"];
>> >      $rootScope.LOG_LEVEL_TO_BOOTSTRAP = ["primary", "info", "warning",
>> >                                           "danger"];
>> > +    var getData = function (url) {
>> > +      return $http.get(url);
>> > +    };
>>
>> How is this better that raw $http.get(url)?
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/coala-analyzer/coala-html/pull/52/files/b7f706d2171c77e341e20b5b680c0c87b1b38b7e#r65679661>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/AHDgOYIINWsziJFcvJgu0GgE60gQeRhgks5qH_L9gaJpZM4IslPq>
>> .
>>
>
>
>
> --
> Best regards,
> Tushar Gautam
> http://gtushar.co
>



-- 
Best regards,
Tushar Gautam
http://gtushar.co

Comment (tushar-rishav) [inline coalahtml/_coalahtml/app/app.js]: I followed http://stackoverflow.com/a/29849524/3673031
ᐧ

On Fri, Jun 3, 2016 at 3:07 PM, Tushar Gautam <tushar.rishav@gmail.com>
wrote:

> I needed to have those data files loaded, before I could actually use them
> later. So I made it synchronous.
> ᐧ
>
> On Fri, Jun 3, 2016 at 3:06 PM, Tushar Gautam <tushar.rishav@gmail.com>
> wrote:
>
>> So this is synchronous. It returns a promise actually.
>> ᐧ
>>
>> On Fri, Jun 3, 2016 at 2:55 PM, Attila <notifications@github.com> wrote:
>>
>>> In coalahtml/_coalahtml/app/app.js
>>> <https://github.com/coala-analyzer/coala-html/pull/52#discussion_r65679661>
>>> :
>>>
>>> >      $rootScope.LOG_LEVELS = ["DEBUG", "INFO", "WARNING", "ERROR"];
>>> >      $rootScope.RESULT_SEVERITY = ["INFO", "NORMAL", "MAJOR"];
>>> >      $rootScope.SEVERITY_TO_BOOTSTRAP = ["info", "warning", "danger"];
>>> >      $rootScope.LOG_LEVEL_TO_BOOTSTRAP = ["primary", "info", "warning",
>>> >                                           "danger"];
>>> > +    var getData = function (url) {
>>> > +      return $http.get(url);
>>> > +    };
>>>
>>> How is this better that raw $http.get(url)?
>>>
>>> —
>>> You are receiving this because you authored the thread.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/coala-analyzer/coala-html/pull/52/files/b7f706d2171c77e341e20b5b680c0c87b1b38b7e#r65679661>,
>>> or mute the thread
>>> <https://github.com/notifications/unsubscribe/AHDgOYIINWsziJFcvJgu0GgE60gQeRhgks5qH_L9gaJpZM4IslPq>
>>> .
>>>
>>
>>
>>
>> --
>> Best regards,
>> Tushar Gautam
>> http://gtushar.co
>>
>
>
>
> --
> Best regards,
> Tushar Gautam
> http://gtushar.co
>



-- 
Best regards,
Tushar Gautam
http://gtushar.co

Comment (tushar-rishav) [inline coalahtml/_coalahtml/app/app.js]: Oh nvm. My bad. I misinterpreted your comment.
ᐧ

On Fri, Jun 3, 2016 at 3:10 PM, Tushar Gautam <tushar.rishav@gmail.com>
wrote:

> I followed http://stackoverflow.com/a/29849524/3673031
> ᐧ
>
> On Fri, Jun 3, 2016 at 3:07 PM, Tushar Gautam <tushar.rishav@gmail.com>
> wrote:
>
>> I needed to have those data files loaded, before I could actually use
>> them later. So I made it synchronous.
>> ᐧ
>>
>> On Fri, Jun 3, 2016 at 3:06 PM, Tushar Gautam <tushar.rishav@gmail.com>
>> wrote:
>>
>>> So this is synchronous. It returns a promise actually.
>>> ᐧ
>>>
>>> On Fri, Jun 3, 2016 at 2:55 PM, Attila <notifications@github.com> wrote:
>>>
>>>> In coalahtml/_coalahtml/app/app.js
>>>> <https://github.com/coala-analyzer/coala-html/pull/52#discussion_r65679661>
>>>> :
>>>>
>>>> >      $rootScope.LOG_LEVELS = ["DEBUG", "INFO", "WARNING", "ERROR"];
>>>> >      $rootScope.RESULT_SEVERITY = ["INFO", "NORMAL", "MAJOR"];
>>>> >      $rootScope.SEVERITY_TO_BOOTSTRAP = ["info", "warning", "danger"];
>>>> >      $rootScope.LOG_LEVEL_TO_BOOTSTRAP = ["primary", "info", "warning",
>>>> >                                           "danger"];
>>>> > +    var getData = function (url) {
>>>> > +      return $http.get(url);
>>>> > +    };
>>>>
>>>> How is this better that raw $http.get(url)?
>>>>
>>>> —
>>>> You are receiving this because you authored the thread.
>>>> Reply to this email directly, view it on GitHub
>>>> <https://github.com/coala-analyzer/coala-html/pull/52/files/b7f706d2171c77e341e20b5b680c0c87b1b38b7e#r65679661>,
>>>> or mute the thread
>>>> <https://github.com/notifications/unsubscribe/AHDgOYIINWsziJFcvJgu0GgE60gQeRhgks5qH_L9gaJpZM4IslPq>
>>>> .
>>>>
>>>
>>>
>>>
>>> --
>>> Best regards,
>>> Tushar Gautam
>>> http://gtushar.co
>>>
>>
>>
>>
>> --
>> Best regards,
>> Tushar Gautam
>> http://gtushar.co
>>
>
>
>
> --
> Best regards,
> Tushar Gautam
> http://gtushar.co
>



-- 
Best regards,
Tushar Gautam
http://gtushar.co

File: coalahtml/_coalahtml/app/controllers/logs.js [hunk 3]
```diff
@@ -1,16 +1,12 @@
 'use strict';
 
 angular.module('coalaHtmlApp')
-  .controller('LogsCtrl', function ($scope, $http) {
+  .controller('LogsCtrl', function ($scope, $rootScope) {
     var parseCoalaProject = function(projectDir, coalaJSON) {
-      coalaJSON = coalaJSON || projectDir + $scope.Constants.coala;
+      coalaJSON = coalaJSON || projectDir + $rootScope.CONSTANTS.coala;
```
Comment (Uran198) [inline coalahtml/_coalahtml/app/controllers/logs.js]: It seems you don't use coalaJSON anymore.

File: coalahtml/_coalahtml/app/controllers/main.js [hunk 4]
```diff
@@ -1,41 +1,32 @@
...
+          }
+        }
+        $scope.data.push(knownFiles);
       };
-    $http.get("data/Constants.json").success(function(constants) {
-      $scope.Constants = constants;
-      parseCoalaProject($scope.Constants.data);
-    });
+
+    parseCoalaProject();
```
Comment (Uran198) [inline coalahtml/_coalahtml/app/controllers/main.js]: Why do you need create a function and call it straight away. Is this function is used (will be used) anywhere else?

**Comment by Uran198** [comment]
ack 34a6ce3b1088ea59ff5594d833e1fe096adfe88d

File: coalahtml/_coalahtml/app/controllers/main.js (hunk 4 above)
Comment (tushar-rishav) [inline coalahtml/_coalahtml/app/controllers/main.js]: I had similar opinion. But I didn't remove the function because in future
we can easily separate the code that need not belong to the function. So we
might need it?
ᐧ

On Fri, Jun 3, 2016 at 3:28 PM, Attila <notifications@github.com> wrote:

> In coalahtml/_coalahtml/app/controllers/main.js
> <https://github.com/coala-analyzer/coala-html/pull/52#discussion_r65683873>
> :
>
> >        };
> > -    $http.get("data/Constants.json").success(function(constants) {
> > -      $scope.Constants = constants;
> > -      parseCoalaProject($scope.Constants.data);
> > -    });
> > +
> > +    parseCoalaProject();
>
> Why do you need create a function and call it straight away. Is this
> function is used (will be used) anywhere else?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-html/pull/52/files/b7f706d2171c77e341e20b5b680c0c87b1b38b7e#r65683873>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHDgOYXslWGacCM1B5HTjQkbxeFgGw3Qks5qH_qpgaJpZM4IslPq>
> .
>



-- 
Best regards,
Tushar Gautam
http://gtushar.co

File: coalahtml/_coalahtml/app/controllers/logs.js (hunk 3 above)
Comment (tushar-rishav) [inline coalahtml/_coalahtml/app/controllers/logs.js]: Indeed
ᐧ

On Fri, Jun 3, 2016 at 3:23 PM, Attila <notifications@github.com> wrote:

> In coalahtml/_coalahtml/app/controllers/logs.js
> <https://github.com/coala-analyzer/coala-html/pull/52#discussion_r65683259>
> :
>
> >      var parseCoalaProject = function(projectDir, coalaJSON) {
> > -      coalaJSON = coalaJSON || projectDir + $scope.Constants.coala;
> > +      coalaJSON = coalaJSON || projectDir + $rootScope.CONSTANTS.coala;
>
> It seems you don't use coalaJSON anymore.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-html/pull/52/files/b7f706d2171c77e341e20b5b680c0c87b1b38b7e#r65683259>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHDgOdI4poyEvrLEnMfv6df5h-hmsTcKks5qH_m0gaJpZM4IslPq>
> .
>



-- 
Best regards,
Tushar Gautam
http://gtushar.co

File: coalahtml/_coalahtml/app/controllers/main.js (hunk 4 above)
Comment (Uran198) [inline coalahtml/_coalahtml/app/controllers/main.js]: I don't think it's very probable that you'll want to parse coala project more than once, but if you'll more functions here, this way it'll be easier to read, so I think it can stay as it is.

File: coalahtml/_coalahtml/app/controllers/logs.js [hunk 5]
```diff
@@ -1,16 +1,10 @@
...
 
 angular.module('coalaHtmlApp')
-  .controller('LogsCtrl', function ($scope, $http) {
-    var parseCoalaProject = function(projectDir, coalaJSON) {
-      coalaJSON = coalaJSON || projectDir + $scope.Constants.coala;
-      $scope.data = [];
-      $http.get(coalaJSON).success(function(response){
-        $scope.data = response.logs;
-      });
+  .controller('LogsCtrl', function ($scope, $rootScope) {
```
Comment (Uran198) [inline coalahtml/_coalahtml/app/controllers/logs.js]: Here for sure you can use di: https://docs.angularjs.org/guide/di

File: coalahtml/_coalahtml/app/app.js (hunk 1 above)
Comment (Uran198) [inline coalahtml/_coalahtml/app/app.js]: You can use list notation here: https://docs.angularjs.org/guide/di

File: coalahtml/_coalahtml/app/controllers/main.js [hunk 6]
```diff
@@ -1,41 +1,36 @@
...
-                });
-            };
-            for (var section in coala_json.results) {
-              if (coala_json.results[section].length > 0) {
-                coala_json.results[section].forEach(parseResult);
+  .controller('MainCtrl', function ($scope, $rootScope, $http) {
+    var parseCoalaProject = function() {
+        $scope.data = [];
+        var knownFiles = {};
+        var parseResult = function(el) {
```
Comment (Uran198) [inline coalahtml/_coalahtml/app/controllers/main.js]: `el` is not very meaningful name. If you're parsing result, you can name argument accordingly.

File: coalahtml/_coalahtml/app/controllers/main.js [hunk 7]
```diff
@@ -1,41 +1,36 @@
...
+              "origin":   el.origin,
+              "severity": el.severity
+            });
+          });
+        };
+        $http.get($rootScope.CONSTANTS.data + $rootScope.CONSTANTS.coala)
+          .then(function(coala_json) {
+            $rootScope.COALA_JSON = coala_json.data;
+            for (var section in $rootScope.COALA_JSON.results) {
+              if ($rootScope.COALA_JSON.results[section].length > 0) {
```
Comment (Uran198) [inline coalahtml/_coalahtml/app/controllers/main.js]: You don't need this check. `forEach` will do nothing on empty array.

**Comment by Uran198** [comment]
It's OK to have no reference to the issue in commit message, if it doesn't reference any issue, and have these commits in one PR. But, IMO, it's not good idea to reference issue if it has nothing to do to the changes. For example, 739ddeec8a6a31a6dc475d136a4833e8a0c2ec8f has nothing to do, already, with adding counts to tabs.

**Comment by Uran198** [comment]
Do you plan on writing tests for this in the following PR? Also, tests are failing, maybe [this](http://stackoverflow.com/questions/17370427/loading-a-mock-json-file-within-karmaangularjs-test) can be helpful.

**Comment by Uran198** [comment]
0276985f1be9c816e7d8b6145eb447f913ff6ab8 should be merged with 71f981c, as it's were tests break.

File: coalahtml/_coalahtml/app/controllers/files.js [hunk 8]
```diff
@@ -1,51 +1,36 @@
...
-              var result = "";
-              fileData[fileName].forEach(function(line){
-                result += line;
-              });
-              $scope.fileContents = result;
-            });
+      if ($rootScope.FILES.hasOwnProperty(fileName)) {
+        $scope.fileType = "dir";
+        $scope.fileContents = {};
+        for(var i_file = 0;i_file < $rootScope.FILES[fileName].length;i_file += 1) {
```
Comment (Uran198) [inline coalahtml/_coalahtml/app/controllers/files.js]: `forEach` would look nicer here

File: coalahtml/_coalahtml/app/views/files.html [hunk 9]
```diff
@@ -1,7 +1,11 @@
 <div class="files">
   <div class="dir-contents" ng-if="fileType === 'dir'">
     <h3>
-    Contents of <span class="filename">{{ fileName }}</span>
+      Contents of
+      <span ng-repeat= "stop in range(len(ROOTHOME), len(fileName) )" class="filename">
+        <a href="#/file/{{ _slice(fileName, 0, stop).join('/') }}">
+          /{{ basename(_slice(fileName, 0, stop).join('/')) }}</a>
+      </span>
```
Comment (Uran198) [inline coalahtml/_coalahtml/app/views/files.html]: You can make one function that will accept `fileName` as an argument and will return pair of link and respected name to be shown. As far as I know, you should not have any logic in your view. Then here you'll only iterate over the output of that function.

File: .coafile [hunk 10]
```diff
@@ -10,14 +10,14 @@ max_line_length = 90
...
-bears = LineLengthBear, SpaceConsistencyBear, CSSLintBear
+bears = LineLengthBear, SpaceConsistencyBear
 use_spaces = true
 
 [html]
 files = coalahtml/_coalahtml/index.html, coalahtml/_coalahtml/app/views/*.html
 bears = LineLengthBear, SpaceConsistencyBear
 use_spaces = true
-max_line_length = 90
+max_line_length = 100
```
Comment (Uran198) [inline .coafile]: unrelated change to this commit.

**Comment by Uran198** [comment]
In b743312defb02b4dc2ff9d9fc83b26ae0f3b109d
`CSSLintBear isn't essential here. Sometimes It returns stupid warnings.` doesn't seems to be a good or informative reason for why you remove this linter.

File: coalahtml/_coalahtml/app/controllers/files.js [hunk 11]
```diff
@@ -33,7 +33,27 @@ angular.module('coalaHtmlApp')
...
+    };
+    $scope.basename = function(path_name) {
+      return getValidPath(path_name).split('/').slice(-1)[0];
+    };
+    var _slice = function(path_name, start, stop) {
+      return getValidPath(path_name).split('/').slice(start, stop);
+    };
+    var len = function(path_name) {
+      return getValidPath(path_name).split('/').length;
+    };
```
Comment (Uran198) [inline coalahtml/_coalahtml/app/controllers/files.js]: Do you really need all these functions? You use them only once. Also you reuse `getValidPath(path_name).split('/')` a lot. So making this only into variable would make things much more simpler.

Comment (tushar-rishav) [inline coalahtml/_coalahtml/app/controllers/files.js]: Um actually I think I am going to use those. I am implementing more changes. Keeping it for now?

File: coalahtml/_coalahtml/app/controllers/files.js [hunk 12]
```diff
@@ -13,12 +13,11 @@ angular.module('coalaHtmlApp')
...
       if ($rootScope.FILES.hasOwnProperty(fileName)) {
         $scope.fileType = "dir";
-        $scope.fileContents = {};
-        for(var i_file = 0;i_file < $rootScope.FILES[fileName].length;i_file += 1) {
-          var dirFile = $rootScope.FILES[fileName][i_file];
-          var dirFileType = $rootScope.FILES.hasOwnProperty(dirFile) ? "dir": "file";
-          $scope.fileContents[dirFile] = dirFileType;
-        }
+        $scope.fileContents = [];
+        $rootScope.FILES[fileName].forEach(function(file){
```
Comment (Uran198) [inline coalahtml/_coalahtml/app/controllers/files.js]: `forEach` should be added in dde0dd67ae29e7a04dafbf5b6410f5f1a5447489. Here it's unrelated.

File: coalahtml/_coalahtml/app/controllers/files.js [hunk 13]
```diff
@@ -33,7 +33,28 @@ angular.module('coalaHtmlApp')
...
+    };
+    $scope.basename = function(path_name) {
+      return getValidPath(path_name).slice(-1)[0];
+    };
+    var _slice = function(path_name, start, stop) {
+      return getValidPath(path_name).slice(start, stop);
+    };
+    var len = function(path_name) {
+      return getValidPath(path_name).length;
+    };
```
Comment (Uran198) [inline coalahtml/_coalahtml/app/controllers/files.js]: You have a whole bunch of functions that basically call one method. Also, you split path_name every time so it'll have slightly worse performance.

Comment (Uran198) [inline coalahtml/_coalahtml/app/controllers/files.js]: Especially in `len` You call this function in for loop, so it'll get called every time and every time, it'll split the path_name.

**Comment by Uran198** [comment]
ack 26668908e780e633464766e569370e4f0a15cfc9 70f72e104829325be58516231a7267de45010077 ec05b1700e16904297a69fa515f6c37357b144b4 d4e901ec11bcdb5afb0acba67a9ffb34deecdea8 9758e1cd16702d504fd5893e5dbe5611007d6c29 5462509acc4ed0d73c39c50f8d08cd53ae35ac77

File: coalahtml/_coalahtml/app/controllers/files.js [hunk 14]
```diff
@@ -33,7 +33,31 @@ angular.module('coalaHtmlApp')
...
+      path_name = path_name || $rootScope.ROOTHOME;
       return path_name.split('/').slice(-1)[0];
     };
+    var len = function(path_name) {
+      path_name = path_name || $rootScope.ROOTHOME;
+      return path_name.split('/').length;
+    };
+    $scope.getNameLink = function(fileName) {
+      var linkNameContainer = [], files = fileName.split('/');
+      var base = files.slice(0, len($rootScope.ROOTHOME)).join('/');
```
Comment (Uran198) [inline coalahtml/_coalahtml/app/controllers/files.js]: Isn't `files.slice(0, len($rootScope.ROOTHOME)).join('/')` the same as `$rootScope.ROOTHOME`?

File: coalahtml/_coalahtml/app/controllers/files.js [hunk 15]
```diff
@@ -33,7 +33,31 @@ angular.module('coalaHtmlApp')
...
+      var prev = {
+        'name' : '/',
+        'link' : "#/file/" + base
+      };
+      linkNameContainer.push(prev);
+      var fileNameLen = len(fileName), rootHomeLen = len($rootScope.ROOTHOME);
+      for (var index = rootHomeLen; index < fileNameLen; index++) {
+        linkNameContainer.push({
+          'name' : index < fileNameLen-1 ? files[index] + prev.name : files[index],
```
Comment (Uran198) [inline coalahtml/_coalahtml/app/controllers/files.js]: `prev.name` is always '/', so why not use '/' here? Also, maybe something like this would look nicer:
```
'name' : files[index] + (index < fileNameLen-1 ? '/' : ''),
```

File: coalahtml/_coalahtml/app/controllers/files.js [hunk 16]
```diff
@@ -33,7 +33,31 @@ angular.module('coalaHtmlApp')
...
+          'link' : prev.link + '/' + files[index]
+        });
+        prev.link = prev.link + '/' + files[index];
```
Comment (Uran198) [inline coalahtml/_coalahtml/app/controllers/files.js]: You can modify `prev.link` before push and then push it. Then there will be no two: `prev.link + '/' + files[index]` pieces.

File: coalahtml/_coalahtml/app/controllers/files.js [hunk 17]
```diff
@@ -33,7 +33,31 @@ angular.module('coalaHtmlApp')
     };
     $scope.fileName = $routeParams.fileName || $rootScope.ROOTHOME;
     parseCoalaProject($scope.fileName);
-    $scope.basename = function(path_name){
+
+    $scope.basename = function(path_name) {
+      path_name = path_name || $rootScope.ROOTHOME;
```
Comment (Uran198) [inline coalahtml/_coalahtml/app/controllers/files.js]: Why this change?

File: coalahtml/_coalahtml/app/controllers/files.js [hunk 18]
```diff
@@ -33,7 +33,31 @@ angular.module('coalaHtmlApp')
...
       return path_name.split('/').slice(-1)[0];
     };
+    var len = function(path_name) {
+      path_name = path_name || $rootScope.ROOTHOME;
+      return path_name.split('/').length;
+    };
```
Comment (Uran198) [inline coalahtml/_coalahtml/app/controllers/files.js]: IMO there is no need in this function, also name is ambiguous.

File: coalahtml/_coalahtml/app/controllers/files.js (hunk 17 above)
Comment (tushar-rishav) [inline coalahtml/_coalahtml/app/controllers/files.js]: `path_name` is `undefined` . It happens the first time page file tab is loaded. There is no fileName passed. For this case default is root.

File: coalahtml/_coalahtml/app/controllers/files.js (hunk 18 above)
Comment (tushar-rishav) [inline coalahtml/_coalahtml/app/controllers/files.js]: I am calculating the same thing (https://github.com/coala-analyzer/coala-html/pull/52/commits/4922d4c32e9d92ba2d9764ef3f6a3cdb035d93b9#diff-b803c22fe9d790db802829435dd58480R53)[2 times]. To make the `getNameLink` clean, I need to separate the process. 
I agree, the name is ambiguous. I will change the name.

Comment (Uran198) [inline coalahtml/_coalahtml/app/controllers/files.js]: You split path every time, this is also not good.

12:04, пт, 10.06.2016 Tushar <notifications@github.com>:

Comment (tushar-rishav) [inline coalahtml/_coalahtml/app/controllers/files.js]: nvm ^

**Comment by Uran198** [comment]
ack 8eea9cb9bbb64f9497ef64cce0e6ad02ea36f875 b785a1ae09e615ffe59a39c0936999828ef566f2 f7d85e5a8575d56a0fd7117ff5cb22cc51ca331c

**Comment by Uran198** [comment]
ack 67a6f6b07992c2720523d33a2e995ac9f5b37f46 19e155091d90a110dacd9a7a78f03e3f2f26756d d7db2a7437b95ec4bb7f6f64d157490eb8130a99 e666514d0c17f71314c3976b9970ed093999cf21 fdc040d6aaeefe24945ba64203329e959bc640a9 c6a75069fbc17edd1dfc21c4e0431646a508134b d992273384f3f6b7f6bbd6227871abf17de24565 944b3cf491cca4163b4598d96b1f82516bab6cf8 13a78cd6ed7f1d5e02875d9fcbf997a1c3f43a04 961881e95a76a2ad17ea6485e78dcf50ddb1d77b

**Comment by Uran198** [comment]
@rultor merge
